dts: Fix some warnings when we build dts files

galak opened the pull request.

galak committed:
boards: arc: nsim: cleanup dts warnings

Fix the following device tree warnings:

unit-address and first reg (0x40000) don't match for iccm@0
unit-address and first reg (0x80040000) don't match for dccm@80000000

Re-work iccm and dccm reg address and size to be based on #defines.
This allows the nsim_sem_normal.dts to override defaults that are set in
nsim.dtsi.  Utilize DT_ADDR macro to take a 'unit-address' still value
and convert it into a hex value (just prepending 0x).

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
boards: xtensa: fix device tree build warnings

Fix warnings of the following form:

unit-address and first reg (0x30000) don't match for partition@10000
unit-address and first reg (0x200000) don't match for partition@1F0000

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
dts: introduce macro's to help with changing unit-addr

Since there are times that we include a base .dtsi file and change the
reg address we tend to end up with warnings that the reg address and
unit-address don't match.  To handle this introduce a simple DT_ADDR(x)
macro that will prepend '0x' to a 'unit-address' style address.

  #define DCCM_ADDR 80000000 /* in unit-address format */

  dccm0: dccm@DCCM_ADDR {
     reg = <DT_ADDR(DCCM_ADDR) 0x1>;
     ...
  };

This allows the dts file to override the value of DCCM_ADDR and than to
have the unit-address of the node and the reg address match.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

dcpleung reviewed (approved):
LGTM on the intel_s1000_crb change.

galak committed:
boards: em_starterkit: cleanup dts warnings

Fix the following device tree warnings:

unit-address and first reg (0x20000) don't match for iccm@0
unit-address and first reg (0x80010000) don't match for dccm@80000000

Since the em_starterkit_em7d_normal has a different base address for
iccm & dccm, and most of the em_starterkit variants have different sizes
for iccm & dccm.  Just define the nodes in the specific
em_starterkit*.dts file and remove them from emsk.dtsi.  This removes
the issue reported in the warning.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
